test/standalone/test-index.js

```diff
@@ -164,6 +164,36 @@ describe('index.js', function() {
     });
   });
 
+  describe('labels', function(){
+    it('should add labels to spans', function() {
+      agent.start();
+      cls.getNamespace().run(function() {
+        agent.private_().createRootSpanData('root', 1, 2);
+        var spanData = agent.startSpan('sub', {test1: 'value'});
+        agent.endSpan(spanData);
+        var traceSpan = spanData.span;
+        assert.equal(traceSpan.name, 'sub');
+        assert.ok(traceSpan.labels);
+        assert.equal(traceSpan.labels.test1, 'value');
+      });
+    });
+
+    it('should ignore non-object labels', function() {
+      agent.start();
+      cls.getNamespace().run(function() {
+        agent.private_().createRootSpanData('root', 1, 2);
+        var stringWithProperty = 'foo';
+        var spanData = agent.startSpan('sub', 'foo');
+        agent.endSpan(spanData);
+        var traceSpan = spanData.span;
+        assert.equal(traceSpan.name, 'sub');
+        assert.ok(traceSpan.labels);
+        assert.strictEqual(traceSpan.labels[0], undefined);
```

Comment:
Length check would be quite brittle. By default we include a stack-trace property on labels, but checking against 1 would put a tight dependency on the default labels that the agent adds. I do also want a stronger check here, but haven't figured out a better one yet. Suggestions welcome.